gcp.go

```diff
@@ -180,29 +209,81 @@ func (l *gcpLogger) Errorf(ctx context.Context, format string, v ...any) {
 	l.log(ctx, logging.Error, fmt.Sprintf(format, v...))
 }
 
-func (l *gcpLogger) log(ctx context.Context, severity logging.Severity, p any) {
+// AddRequestAttribute adds an attribute (key, value) for the parent request log
+// If the key matches a reserved key, it will be prefixed with "custom_"
+// If the key already exists, its value is overwritten
+func (l *gcpLogger) AddRequestAttribute(key string, value any) {
+	if slices.Contains(l.rsvdKeys, key) {
+		key = customPrefix + key
+	}
+
 	l.mu.Lock()
-	if l.maxSeverity < severity {
-		l.maxSeverity = severity
+	defer l.mu.Unlock()
+	l.reqAttributes[key] = value
+}
+
+// WithAttributes returns an attributer that can be used to add child (trace) log attributes
+func (l *gcpLogger) WithAttributes() attributer {
+	attrs := make(map[string]any)
+	for k, v := range l.attributes {
+		attrs[k] = v
 	}
-	l.logCount++
-	l.mu.Unlock()
 
-	if err, ok := p.(error); ok {
-		p = err.Error()
+	return &gcpAttributer{logger: l, attributes: attrs}
+}
```

Comment:
I know that I have already expressed my opinion that we should not be taking in a key/value pair here, but wanted to point out that we have duplicated the business logic here and in `gcpAttributer.AddAttribute()`

Reply:
Disregard